### ISSUE TITLE
Is the output of this model not the three sub-regions of the tumor(i.e. whole tumor, enhance tumor and tumor core)?

### ISSUE THREAD

**Issue by joe1chief** [IAmSuyogJadhav/3d-mri-brain-tumor-segmentation-using-autoencoder-regularization#14]
In function "preprocess_label", you converts the label into three channels,  representing "ncr", "ed" and "et". However,  the paper mentions that "We also output all 3 nested tumor subregions directly after the sigmoid (instead of using several networks or the softmax over the number of classes). "

By the way, can you tell me the final value of the loss after the model converges? In my experiments, the loss is approaching -0.20 and the dice score is 0.4443 after 300 epochs (the patch size set to 32, 192, 160).

**Comment by Infinity0106** [comment]
I'm also having the same problem in training with BRATS 2018 but I can't make it converge

**Comment by IAmSuyogJadhav** [comment]
The label format is correct as per the paper. The three output channels represent segmentation masks for "ncr", "ed" and "et" respectively. The ground truth has only one channel, with each label (ncr, ed, et) represented by a different number. It needs to be separated into 3 different channels to make it work for sigmoid.

The final value of the loss can get negative, that is totally fine. The loss function has a higher weight for the negative of dice coefficient term and positive weights for reconstruction loss. Thus, the total loss can get negative.
